### PR TITLE
[BISERVER-13909] - Uploaded PDI Job/Transformation has file size of 0 on the Properties screen

### PR DESCRIPTION
**Pull request by joana-fb**
@pentaho/tatooine_dev

### CODE REVIEW

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

File: plugins/pur/core/src/main/java/com/pentaho/repository/importexport/StreamToJobNodeConverter.java [hunk 1]
```diff
@@ -146,10 +147,18 @@ Repository connectToRepository() throws KettleException {
    */
   public IRepositoryFileData convert( final InputStream inputStream, final String charset, final String mimeType ) {
     try {
+      InputStream newInputStream = inputStream;
       long size = inputStream.available();
+
+      if ( size == 0 ) {
+        byte[] fileContent = IOUtils.toByteArray( newInputStream );
```
Comment (tgf) [inline plugins/pur/core/src/main/java/com/pentaho/repository/importexport/StreamToJobNodeConverter.java:154]: commons-io has a CountingInputStream wrapper that could give a count at the end without having to hold everything in memory

Comment (joana-fb) [reply plugins/pur/core/src/main/java/com/pentaho/repository/importexport/StreamToJobNodeConverter.java]: Thank you for your feedback! CountingInputStream is deprecated, so I used BoundedInputStream

**Comment by buildguy** [comment]
### :x: Build failed in 2m 18s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl plugins/pur/core
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Review by smmribeiro — APPROVED** [review]
Looks good!

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 27m 26s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl plugins/pur/core
```
---
:ok_hand: All tests passed!
 
Tests run: **277**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9682/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**